elastic/kibana
# [canvas] fix parent API should implement unified search interface with react embeddables

Closes https://github.com/elastic/kibana/issues/183766

### test instructions
1. start kibana with `yarn start --run-examples`
2. install sample web logs data set
3. create new canvas work pad
4. Click "Add element" -> "Filter" -> "Dropdown select"
5. Select "data" tab and configure filter to pull from logs sample data and `machine.os.keyword` field.
6. Select "display" tab and "Value column" and "Filter column" to `machine.os.keyword`. 
7. Click "Add element" -> "Filter" -> "Time filter"
9. Click "Select type" -> "Search example"
10. Click panel context menu -> "Settings" and turn off custom time range
11. Click "Expression editor" button in lower right corner to open the expression editor for the map and prefix the expression with `kibana | selectFilter | embeddable`
    <img width="500" alt="Screenshot 2024-05-17 at 11 50 42 AM" src="https://github.com/elastic/kibana/assets/373691/453096d5-c254-4f43-a09c-386586aa05a1">
12. Interact with drop down filter and time range filter. Verify Search react embeddable updates as expected
